gtsfm/frontend/matcher/cacher/matcher_cacher.py

```diff
@@ -0,0 +1,181 @@
+"""Decorator for caching matcher output.
```

Comment:
a bit more high-level details would be nice.

url to design pattern? brief usage example?

Reply:
I can add the usage example, but I dont think the design paradigm should go in there. People interested in why it is done this way can refer the PR.

Reply:
I have added the information in file docstring and the readme.